templates/install_puppet.bat.erb

```diff
@@ -0,0 +1,11 @@
+:wait_for_pid
+FOR /F "tokens=*" %%A IN ('tasklist /FI "PID eq <%= @puppet_agent_pid %>"  /NH') DO set _task=%%A
+echo %_task% | findstr "No tasks are running" >nul
+IF NOT %errorlevel% == 0 ( GOTO wait_for_pid )
+
+
+msiexec.exe /qn /i <%= @_msi_location %> /l*v <%= @_logfile %>
```

Comment:
In case the location has spaces (not likely for most cases, but still plausible if unc), you should surround the msi location with quotes.